projects/epbs-implementation-in-prysm.md

```diff
@@ -0,0 +1,113 @@
+
+EIP-7732 introduces a trust-free method for this exchange, ensuring that an honest proposer gets paid and the honest builder’s payload becomes the canonical head of the chain, regardless of the other’s actions.
+
+Validators have a short window (4 seconds on Ethereum mainnet) to validate both consensus and execution state transitions, check blob data availability, and evaluate the new head of the blockchain. By separating the validation tasks, validators only need to perform the consensus state transition immediately, deferring execution and data availability validation to later.
+
+This change allows for faster network propagation by removing the execution payload from the consensus block. It reduces the risk of reorganizations caused by blob transactions and prevents validators from missing attestations. Additionally, it eliminates the need for trusted middleware in block construction delegation.
+
+<figure>
+
+![ePBS slot](https://hackmd.io/_uploads/rJdlt-sd0.png)
+
+<figcaption>
+
+_ePBS slot from potuz's presentation_
+
+</figcaption>
+</figure>
+
+## Specification
+
+The implementation would follow the spec mentioned in [EIP-7732](https://eips.ethereum.org/EIPS/eip-7732).
+
+The spec introduces a new staked Participant called _Builders_ and new honest validators duties called _payload timeliness attestations_.
+
+The slot in the Ethereum protocol is divided into four intervals:
+
+- Honest validators would gather _signed bids_ (`SignedExecutionPayloadHeader`) from builders and submit their consensus blocks (`SignedBeaconBlock`) with these bids.
+- For the second part, Honest validators submit attestations, similar to the current process.
+- For the 3rd Interval, Aggregators aggregate these attestations, and the builder broadcasts either the full payload or a message indicating they are withholding it (`SignedExecutionPayloadEnvelope`).
+- For the 4th Interval, some validators selected for the new `Payload Timeliness Committee (PTC)` attest to the presence and timeliness of the builder’s payload.
+
+At any given slot, the blockchain’s head status can be:
+
+- A block from a previous slot if the current slot’s proposer did not submit a block.
+- An empty block for the current slot if the proposer submitted a block but the builder did not reveal the payload on time.
+- A full block for the current slot if both the proposer and the builder revealed on time.
+
+There are various pacakages, that would require significant changes including but not limited to sync, core, beacon-apis for builder, forkchoice, engine api and blockchain etc. Additionally, there are numerous helper functions and once a working PoC is implemented we would be moving onto Networking packages for the actual p2p auction and bids' gossiping.
```

Comment:
'spelling error   pacakages' to packages

Reply:
thanks amaan, fixed